### PR TITLE
NAS-120190 / 23.10 / Add simplified API for granting ids access to path

### PR DESCRIPTION
**Pull request by anodos325**
This simplified API is for charts use to allow chart to specify uids / gids that need access to a given host path.

### CODE REVIEW

**Review by Ornias1993 — COMMENTED** [review]
This is a great addition and simplifies our lives a lot :)

obviously it still needs implementation in a way that is useable by App creators, preferably by parsing values.yaml after questions.yaml is parsed (like I did in the new dynamic portals code).

But this is a HUGE step up, and really off-sets iX-Systems as a major player in the Kubernetes space 👍

**Comment by anodos325** [comment]
> This is a great addition and simplifies our lives a lot :)
> 
> obviously it still needs implementation in a way that is useable by App creators, preferably by parsing values.yaml after questions.yaml is parsed (like I did in the new dynamic portals code).
> 
> But this is a HUGE step up, and really off-sets iX-Systems as a major player in the Kubernetes space 👍

Waqar or Rehan will work on the exact implementation details for charts integration in a separate ticket.

File: src/middlewared/middlewared/plugins/filesystem_/acl_linux.py [hunk 1]
```diff
@@ -678,3 +678,165 @@ def setacl(self, job, data):
...
+                'id': entry['id'],
+                'perms': convert_perm(entry['access']),
+                'flags': {'BASIC': 'INHERIT'},
+                'type': 'ALLOW'
+            })
+
+        return acl
+
+    def add_to_acl(self, job, data):
+        if os.listdir(data['path']) and not data['options']['force']:
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/filesystem_/acl_linux.py:819]: This will fail if the path is either empty or does not exist

**Review by sonicaj — COMMENTED** [review]
Good work 👍🏻

**Review by yocalebo — APPROVED** [review]

**Pull request merged**